corehq/apps/geospatial/views.py

```diff
@@ -1,8 +1,65 @@
-from django.http import HttpResponseRedirect
+import json
+import jsonschema
+
+from django.http import HttpResponseRedirect, Http404, HttpResponseBadRequest
+from django.utils.decorators import method_decorator
 from django.utils.translation import gettext_lazy as _
+from dimagi.utils.web import json_response
 
+from corehq import toggles
+from corehq.apps.domain.views.base import BaseDomainView
 from corehq.apps.geospatial.reports import CaseManagementMap
+from .constants import POLYGON_COLLECTION_GEOJSON_SCHEMA
+from .models import GeoPolygon
 
 
 def geospatial_default(request, *args, **kwargs):
     return HttpResponseRedirect(CaseManagementMap.get_url(*args, **kwargs))
+
+
+class GeoPolygonView(BaseDomainView):
+    urlname = 'geo_polygon'
+
+    @method_decorator(toggles.GEOSPATIAL.required_decorator())
```

Comment:
Instead of adding the `@method_decorator(toggles.GEOSPATIAL.required_decorator())` to every method of the view, you can declare the `dispatch` method on the view and simply add it there. The `dispatch` method is like the [entrypoint to the view](https://medium.com/@hassanraza/what-is-dispatch-used-for-in-django-c29af0653e94#:~:text=The%20dispatch%20method%20takes%20in,middleman%20between%20requests%20and%20responses.).

Reply:
True